common/functions/src/scalars/expressions/cast_from_string.rs

```diff
@@ -89,11 +89,30 @@ pub fn cast_from_string(
 #[inline]
 pub fn string_to_timestamp(date_str: impl AsRef<[u8]>, tz: &Tz) -> Option<DateTime<Tz>> {
     let s = std::str::from_utf8(date_str.as_ref()).ok();
-    s.and_then(|c| tz.datetime_from_str(c, "%Y-%m-%d %H:%M:%S%.f").ok())
+    if let Some(s) = s {
+        // convert zero timestamp to `1970-01-01 00:00:00`
+        if s.len() >= 19 && &s[..19] == "0000-00-00 00:00:00" {
```

Comment:
Do we only do this hack for `0000-00-00 00:00:00`? Or can we do this for all time before `1970-01-01 00:00:00`?

Reply:
As https://github.com/datafuselabs/databend/issues/6577  described, we should cover the case of time before `1000-01-01 00:00:00` into `1000-01-01 00:00:00`